platform/source/mbed_retarget.cpp

```diff
@@ -391,9 +473,11 @@ static int reserve_filehandle()
 
     return fh_i;
 }
+#endif // !(MBED_CONF_PLATFORM_STDIO_MINIMAL_CONSOLE_ONLY)
 
 int mbed::bind_to_fd(FileHandle *fh)
 {
+#if !(MBED_CONF_PLATFORM_STDIO_MINIMAL_CONSOLE_ONLY)
```

Comment:
Why not remove both declaration & implementation. This is only used in fdopen which could/should be removed as well. fdopen is currently called in Stream but Stream should be compiled out if  MBED_CONF_PLATFORM_STDIO_MINIMAL_CONSOLE_ONLY is used.

Reply:
This was done with the thinking that no other file should be modified as part of this work.

Before any change is made, can you confirm that we should also modify files other than `mbed_retarget.cpp`, `mbed_retarget.h`, and `platform/mbed_lib.json`?

Reply:
This interface 
```
namespace mbed {

/** This call is an analogue to POSIX fdopen().
 *
 *  It associates a C stream to an already-opened FileHandle, to allow you to
 *  use C printf/scanf/fwrite etc. The provided FileHandle must remain open -
 *  it will be closed by the C library when fclose(FILE) is called.
 *
 *  The net effect is fdopen(bind_to_fd(fh), mode), with error handling.
 *
 *  @param fh       a pointer to an opened FileHandle
 *  @param mode     operation upon the file descriptor, e.g., "w+"
 *
 *  @returns        a pointer to FILE
 */
std::FILE *fdopen(mbed::FileHandle *fh, const char *mode);

/** Bind an mbed FileHandle to a POSIX file descriptor
 *
 * This is similar to fdopen, but only operating at the POSIX layer - it
 * associates a POSIX integer file descriptor with a FileHandle, to allow you
 * to use POSIX read/write calls etc. The provided FileHandle must remain open -
 * it will be closed when close(int) is called.
 *
 *  @param fh       a pointer to an opened FileHandle
 *
 *  @return         an integer file descriptor, or negative if no descriptors available
 */
int bind_to_fd(mbed::FileHandle *fh);

} // namespace mbed
```
will not be functional when MBED_CONF_PLATFORM_STDIO_MINIMAL_CONSOLE_ONLY is true. So it should be cleaner to remove it and this will also save memory.

I am not sure what the impact is (we do want to limit the number of files modified in this PR); as far as I can tell it might only be used in `Sream` at the moment and `Stream` is targeted for deprecation.

Reply:
[I only removed the implementation](https://github.com/hugueskamba/mbed-os/compare/93f364edd2488439214bd28a928fcff6910377e7..f3f1c0fdf36f8ca6cf191aa1c815dbeee0787f45).

Removing the declaration causes:
```bash
Compile [ 98.6%]: Stream.cpp
[Error] Stream.cpp@27,30: invalid conversion from 'mbed::Stream*' to 'int' [-fpermissive]
[ERROR] '_queue.SimpleQueue' object has no attribute 'queue'
[mbed] ERROR: "/usr/bin/python3" returned error.
       Code: 1
       Path: "/home/hugkam01/projects/mbed/examples/mbed-os-example-blinky"
       Command: "/usr/bin/python3 -u /home/hugkam01/projects/mbed/examples/mbed-os-example-blinky/mbed-os/tools/make.py -t gcc_ARM -m k64f --source . --build ./BUILD/K64F/GCC_ARM"
       Tip: You could retry the last command with "-v" flag for verbose output
```

Reply:
As per the design document, deprecation of Stream is scheduled as future work. If possible, I do not want to tag this work onto the minimal console work to not open another can of worms at this time.

Reply:
OK.